src/http_utils.cpp

```diff
@@ -230,33 +230,33 @@ std::string http_utils::standardize_url(const std::string& url) {
 }
 
 const std::string http_utils::generate_random_upload_filename(const std::string& directory) {
-    std::string filename = directory + path_separator + upload_filename_template;
+    std::string filename = directory + PATH_SEPARATOR + UPLOAD_FILENAME_TEMPLATE;
     char *template_filename = strdup(filename.c_str());
     int fd = 0;
 
 #if defined(_WIN32)
     if (0 != _mktemp_s(template_filename, filename.size() + 1)) {
         free(template_filename);
-        throw std::exception();
+        throw generateFilenameException();
     }
     if (0 != _sopen_s(&fd, template_filename, _O_CREAT | _O_EXCL | _O_NOINHERIT, _SH_DENYNO, _S_IREAD | _S_IWRITE)) {
         free(template_filename);
-        throw std::exception();
+        throw generateFilenameException();
     }
     if (fd == -1) {
         free(template_filename);
-        throw std::exception();
+        throw generateFilenameException();
     }
     _close(fd);
-#else
+#else  // _WIN32
     fd = mkstemp(template_filename);
 
     if (fd == -1) {
         free(template_filename);
-        throw std::exception();
+        throw generateFilenameException();
     }
     close(fd);
-#endif
+#endif  // _WIN32
     std::string ret_filename = std::string(template_filename);
```

Comment:
If `template_filename` is a `char*` you might not need to wrap it into an `std::string`; the `operator=` does the job already for you.